src/Chapter2.hs

```diff
@@ -605,8 +622,11 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate xs = duplicate' xs []
 
+duplicate' :: [a] -> [a] -> [a]
+duplicate' [] acc = reverse acc
+duplicate' (x:xs) acc = duplicate' xs (x:x:acc)
```

Comment:
Your implementation is correct :+1: But it can be made more performant. `reverse` is not a fast function, since it traverses the whole list. Here it's possible to slightly rewrite the implementation by adding elements to the beginning of the result instead adding to a separate variable `acc`.